tests/protocol/test_wire.py

```diff
@@ -342,7 +342,7 @@ def test_get_in_vm_artifacts_profile_host_ga_plugin(self, *args):
         wire_protocol_client.ext_conf.artifacts_profile_blob = testurl
         goal_state = GoalState(WireProtocolData(DATA_FILE).goal_state)
         wire_protocol_client.get_goal_state = Mock(return_value=goal_state)
-        wire_protocol_client.fetch = Mock(side_effect=[None, '{"onHold": "true"}'.encode('utf-8')])
```

Comment:
We were fooling ourselves with all of these mocks.  We improperly characterized the return type of this method.  It is not bytes, it is a string. 😭